heat/core/operations.py

```diff
@@ -172,8 +172,23 @@ def __local_op(operation, x, out, **kwargs):
         if kwargs_dtype:
             del kwargs['dtype']
         result = operation(x._DNDarray__array.type(torch_type), **kwargs)
-        return dndarray.DNDarray(result, tuple(result.shape), promoted_type if kwargs_dtype is None else kwargs_dtype,
-                                 x.split, x.device, x.comm)
+        split = x.split
+        gshape = list(result.shape)
+        if x.split is not None:
+            # The operation could have reduced the result by one axis
+            axis = x.split if x.split < len(result.shape) else len(result.shape) - 1
+            if len(result.shape) > 0:
+                split_shape = torch.tensor([result.shape[axis]])
+            else:
+                # 0-Dimensional tensor in other words just a number on each process
+                split_shape = torch.tensor([1 if x.shape[x.split] > 0 else 0], dtype=result.dtype)
+                gshape = [0]
+            x.comm.Allreduce(MPI.IN_PLACE, split_shape, MPI.SUM)
```

Comment:
Why not simply use the gshape and lshape of x?

Reply:
In the case where a reduction happened this is necessary.